### ISSUE TITLE
Predictions are switching between results

### ISSUE THREAD

**Issue by simonfranz** [despoisj/DeepAudioClassification#6]
Hi there, 
I've taken the DeepAudioClassification and trained with 10k slices, afterwards improved it with 5k slices and again with 15k slices for each genre.
![training-data](https://i.imgur.com/GoqVp1j.png)

Despite the accuracy shwon above the result shown at the last step was about 98%.

Running the test against the model there was an 28%. 
Can I approve this percentage by training with even bigger data?

What I've done so far:
1. I'm classifying in 23 genres, so there is a lot of data. I have refactored the dataSetTool, because I noticed that pickle  quited every time I wanted to write data > 3 GB. Now  the dataset is splittet in 1GB chunks and is concatenated for training, test, validation.
2. LIke described in the comments in the code the end of the song is discarded. I've modified the slicing that i get the double amount of slices (shifted by desiredSize/2), so I'm discarding a bit of the beginning to get some more of the end. This created the doubled amount of slices which should be fine.

Said this, I've come to the initial problem if I run a song against the created model multiple times, every time I get an other result. Once it's pop, once it's rock, once it's reaggae.

If I create the predictions for each slice seperate I've recognized, that the order of the sliced in the data array is not the same, but this shouldn't have influence to the result, shouldn't it? Also the result's itself does not vary. If I sum up manually (in code with the predictions for each slice) the max is always at pop.
I don't know if this behaviour is "normal", so my next approach would be to drop the predictions for a single array with all slices, and get the predictions on every single slice and sum the predictions manually like described in your medium-post and ignore predictions <70%. 

Do you have any tips on the switching predictions?
Thanks in advance!

**Comment by despoisj** [comment]
Hi Simon,

Thanks for your detailed comment!
I think that your slicing/dataset loading methods are good additions.

The main issues I see are:
• 23 genres might be too much for this model, try to make sure they are roughly in the same proportions (use over/under sampling), and plot a confusion matrix to try to figure out what is going on. The difference between Train/Test results is symptomatic of overfitting, which is normal given a powerful network and enough time. You could also try upsizing the network, while adding more regularization.

• The very weird part is that it shouldn't be random. Once the model is trained, dropout is disabled and the network should give consistent results. You should absolutely investigate that first!

Let me know!

Julien

**Comment by simonfranz** [comment]
Thanks for the answer too.

* Would you accept PR for the sclicing of data-sets?

* I have reduced the number of genres to 10.  The number of slices per genre differ from about 5k slices to up to 72k. So training the model with a ```filesPerGenre``` to 5000 should be good enough. Maybe I will decrease the set of genres even further to have at least 10k per genre. I don't know where to begin to see a plot of the cinfusion matrix. Do you have any hints where I could begin? 

* May I also ask another question, as far as I read the loss function should go down after time. Something like this
![Normal Loss function](http://tflearn.org/img/loss_acc.png)
On the run I noticed that the loss function on this model locks like this. Should the value of the loss-function keep being so high?
![My Loss function](https://i.imgur.com/fgl7Oz8.png)

* After I have read the slices into the vector I have shuffle them (```from random import shuffle```). I thought that the position of the data would not be crucial, but in this case it was. Removing the shuffle of the read slices removed the changing result, so given a file, the result stays the same at different predicitons.

* To get some more information about the learning process I've added ```model = tflearn.DNN(convnet, tensorboard_verbose=3)``` to the model. I think this also would help in the example to see some more details on tensorboard.

Thanks again for the very nice project!
